transform.ts

```diff
@@ -241,6 +240,25 @@ function createInterface(path: NodePath, componentTypes: CollectedTypes) {
 
   return typeName
 }
+/**
+ * If forwardRef is being used, declare the props.
+ * Otherwise, return false
+ */
+function addForwardRefTypes(path: NodePath, typeName: string): boolean {
+  // for `React.forwardRef()`
+  let isForwardRef = path.node.callee?.property?.name === "forwardRef"
+  if (isForwardRef) {
+    path.node.callee.property.name = `forwardRef<React.ReactHTMLElement, ${typeName}>`
```

Comment:
```suggestion
    path.node.callee.property.name = `forwardRef<HTMLElement, ${typeName}>`
```

Reply:
see my notes from the description

> setting the nodes names directly using a template string approach is less than ideal of course, but I couldn't get other approaches working, such as setting typeParameters directly. would be happy to improve this part

I tried setting `path.value.typeArguments` directly, but with identifiers only - I didn't try wrapping them in `j.genericTypeAnnotation(` which makes sense!

Reply:
> maybe I'm working from a different version of @types/react but I don't get an React.HTMLElement only React.ReactHTMLElement?

As I showed in my suggestion, the proper interface is `HTMLElement` without the `React.` prefix.  `HTMLElement` comes from TypeScript (lib.dom.ts) as it describes the actual `HTMLElement` interface as specified in the web spec.

- https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L6357
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement